src/db_adapters/accounts.rs

```diff
@@ -0,0 +1,157 @@
+                                    ))
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect();
+                        Some(accounts)
+                    }
+                    _ => None,
+                }
+            } else {
+                None
+            }
+        })
+        .flatten()
+        .collect();
+
+    loop {
+        match diesel::insert_into(schema::accounts::table)
+            .values(accounts_to_create.clone())
+            .on_conflict_do_nothing()
+            .execute_async(&pool)
+            .await
+        {
+            Ok(_) => break,
+            Err(async_error) => {
+                error!(
+                    target: crate::INDEXER_FOR_EXPLORER,
+                    "Error occurred while Accounts were adding to database. Retrying in {} milliseconds... \n {:#?} \n {:#?}",
+                    crate::INTERVAL.as_millis(),
+                    async_error,
+                    &accounts_to_create,
+                );
+                tokio::time::delay_for(crate::INTERVAL).await;
+            }
+        }
+    }
+}
+
+async fn remove_accounts(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    outcomes: &[near_indexer::IndexerExecutionOutcomeWithReceipt],
+) {
+    eprintln!("{:#?}", &outcomes);
+    let accounts_to_delete: Vec<(String, String)> = outcomes
+        .iter()
+        .filter_map(|outcome_with_receipt| {
+            if let Some(receipt) = &outcome_with_receipt.receipt {
+                match &receipt.receipt {
+                    near_primitives::views::ReceiptEnumView::Action { actions, .. } => {
+                        let accounts: Vec<(String, String)> = actions
+                            .iter()
+                            .filter_map(|action| {
+                                if let near_primitives::views::ActionView::DeleteAccount {
+                                    ..
+                                } = action
+                                {
+                                    Some(
+                                        (
+                                            (&receipt.receiver_id).to_string(),
+                                            receipt.receipt_id.to_string(),
+                                        )
+                                    )
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect();
+                        Some(accounts)
+                    }
+                    _ => None,
+                }
+            } else {
+                None
+            }
+        })
+        .flatten()
+        .collect();
+
+    eprintln!("ACCOUNT_TO_DELETE:\n {:#?}", &accounts_to_delete);
+
+    for (account_id, deleted_by_receipt_id) in accounts_to_delete {
```

Comment:
This is quite inefficient, but DeleteAccounts are rare, so we can live with it as is. Just FYI, you can update multiple records to distict values in a single query: https://stackoverflow.com/a/18799497/1178806